src/Layer/OrientedImageLayer.js

```diff
@@ -205,9 +205,12 @@ class OrientedImageLayer extends GeometryLayer {
      * Delete background, but doesn't delete OrientedImageLayer.material. For the moment, this material visibility is set to false.
      * You need to replace OrientedImageLayer.material applied on each object, if you want to continue displaying them.
      * This issue (see #1018 {@link https://github.com/iTowns/itowns/issues/1018}) will be fixed when OrientedImageLayer will be a ColorLayer.
-     */
-    delete() {
-        super.delete();
+    * @param {boolean} [clearCache=false] Whether to clear the layer cache or not
+    */
+    delete(clearCache) {
+        if (clearCache) {
```

Comment:
Yep I think you're right, I will make the modifications :)

Side question: Regarding the filtering of the geometries by layer id in ObjectRemovalHelper, which usecase does it serve? I cannot think of a case where a GeometryLayer would have geometries that don't have the id of this geometry layer, is there one ?

Reply:
In case a `LabelLayer` is added, the labels are added as children of a tile. Sometimes, the method is called with the tile as object parameter. The filtering prevents removing labels.

Reply:
Done